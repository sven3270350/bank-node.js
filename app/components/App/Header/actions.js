/*
 *
 * HeaderApp actions
 *
 */

import {
  TOGGLE_MENU_DESKTOP,
  TOGGLE_MENU_MOBILE,
  LOGOUT,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  IS_NOTIFICATION,
  NEW_NOTIFICATION,
  UNSET_NOTIFICATION,
  HIDDEN_MENU_MOBILE,
  TOGGLE_NOTIFICATION,
} from './constants';

export function toggleNotificationAction() {
  return {
    type: TOGGLE_NOTIFICATION,
  };
}

export function isNotificationAction() {
  return {
    type: IS_NOTIFICATION,
  };
}

export function newNotificationAction(notificationCount) {
  return {
    type: NEW_NOTIFICATION,
    notificationCount,
  };
}

export function unsetNotificationAction() {
  return {
    type: UNSET_NOTIFICATION,
  };
}

export function toggleMenuDesktopAction() {
  return {
    type: TOGGLE_MENU_DESKTOP,
  };
}

export function toggleMenuMobileAction() {
  return {
    type: TOGGLE_MENU_MOBILE,
  };
}

export function hiddenMenuMobileAction() {
  return {
    type: HIDDEN_MENU_MOBILE,
  };
}

export function logoutAction(id) {
  return {
    type: LOGOUT,
    id,
  };
}

export function successLogoutAction() {
  localStorage.removeItem('id_token');
  return {
    type: LOGOUT_SUCCESS,
  };
}

export function errorLogoutAction(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}
