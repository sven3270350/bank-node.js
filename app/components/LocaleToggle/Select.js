import styled from 'styled-components';
import { LINE_GREY } from 'utils/colors';

const Select = styled.select`
  font-family: 'Lato';
  display: inline-block;
  width: 100%;
  cursor: pointer;
  padding: 8px 10px;
  outline: 0;
  border: 1px solid ${LINE_GREY};
  border-radius: 2px;
  background: #ffffff;
  color: #000000;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  text-indent: 0.01px;
  text-overflow: '';
`;

export default Select;
