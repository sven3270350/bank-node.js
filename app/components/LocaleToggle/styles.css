.select__locale {
  position: relative;
  display: inline-block;
  width: 65px;
}
.select__locale select {
  font-family: 'Lato';
  display: inline-block;
  width: 100%;
  cursor: pointer;
  padding: 8px 10px;
  outline: 0;
  border: 1px solid grey;
  border-radius: 2px;
  background: #ffffff;
  color: #000000;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  text-indent: 0.01px;
  text-overflow: '';
}

.select__locale select:-moz-focusring {
  color: transparent;
  text-shadow: 0 0 0 #000;
}

.select__locale select::-ms-expand {
  display: none;
}

.select__locale select:hover,
.select__locale select:focus {
  color: #000000;
  background: #ffffff;
}

.select__locale select:disabled {
  opacity: 1;
  pointer-events: none;
}

.select_arrow--locale {
  position: absolute;
  top: 11px;
  right: 15px;
  width: 10px;
  height: 10px;
  border: solid #0029ab;
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
  -webkit-transition: 0.3s ease;
  -o-transition: 0.3s ease;
  transition: 0.3s ease;
}

.select__locale select:hover ~ .select_arrow--locale,
.select__locale select:focus ~ .select_arrow--locale {
  border-color: #0029ab;
}

.select__locale select:disabled ~ .select_arrow--locale {
  border-top-color: #0029ab;
}

.select__locale select:focus ~ .select_arrow--locale {
  -webkit-transform: translateY(40%) rotate(225deg);
  -ms-transform: translateY(40%) rotate(225deg);
  transform: translateY(40%) rotate(225deg);
}

.select__locale select:not(:hover) ~ .select_arrow--locale {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}
