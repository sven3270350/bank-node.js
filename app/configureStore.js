/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from 'redux';
import { fromJS } from 'immutable';
import { routerMiddleware } from 'connected-react-router/immutable';
import createSagaMiddleware from 'redux-saga';
import immutableTransform from 'redux-persist-transform-immutable';
import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';
import { persistStore, autoRehydrate } from 'redux-persist-immutable';
import createReducer from './reducers';

const persistConfig = {
  transforms: [immutableTransform()],
  key: 'root',
  storage,
};

const sagaMiddleware = createSagaMiddleware();
const rootReducer = createReducer();
const persistedReducer = persistReducer(persistConfig, rootReducer);

export default function configureStore(initialState = {}, history) {
  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares), autoRehydrate()];

  // If Redux DevTools Extension is installed use it, otherwise use Redux compose
  /* eslint-disable no-underscore-dangle, indent */
  const composeEnhancers =
    process.env.NODE_ENV !== 'production' &&
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose;
  /* eslint-enable */

  const store = createStore(
    persistedReducer,
    fromJS(initialState),
    composeEnhancers(...enhancers),
  );

  // Extensions
  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {}; // Reducer registry
  store.injectedSagas = {}; // Saga registry

  // Make reducers hot reloadable, see http://mxs.is/googmo
  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      store.replaceReducer(createReducer(store.injectedReducers));
    });
  }

  const persistor = persistStore(store);
  return { store, persistor };
}
