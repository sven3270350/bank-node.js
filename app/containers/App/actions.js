/*
 *
 * App actions
 *
 */

import {
  IS_LOGGED,
  LOGGED_IN,
  LOGOUT,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  ENQUEUE_SNACKBAR,
  CLOSE_SNACKBAR,
  REMOVE_SNACKBAR,
  TOGGLE_NAVIGATION_DESKTOP,
  TOGGLE_NAVIGATION_MOBILE,
  TOGGLE_MESSAGES,
  TOGGLE_NOTIFICATIONS,
  CHECK_NEW_MESSAGES,
  CHECK_NEW_MESSAGES_SUCCESS,
  CHECK_NEW_MESSAGES_ERROR,
  CHECK_NEW_NOTIFICATIONS,
  CHECK_NEW_NOTIFICATIONS_SUCCESS,
  CHECK_NEW_NOTIFICATIONS_ERROR,
  GET_NEW_NOTIFICATIONS,
  GET_NEW_NOTIFICATIONS_SUCCESS,
  GET_NEW_NOTIFICATIONS_ERROR,
  UNSET_NEW_NOTIFICATIONS,
  UNSET_NEW_NOTIFICATIONS_SUCCESS,
  UNSET_NEW_NOTIFICATIONS_ERROR,
} from './constants';

export function checkNewMessagesAction() {
  return {
    type: CHECK_NEW_MESSAGES,
  };
}

export function checkNewMessagesSuccessAction(messages) {
  return {
    type: CHECK_NEW_MESSAGES_SUCCESS,
    messages,
  };
}

export function checkNewMessagesErrorAction(error) {
  return {
    type: CHECK_NEW_MESSAGES_ERROR,
    error,
  };
}

export function checkNewNotificationsAction() {
  return {
    type: CHECK_NEW_NOTIFICATIONS,
  };
}

export function checkNewNotificationsSuccessAction(notificationCount) {
  return {
    type: CHECK_NEW_NOTIFICATIONS_SUCCESS,
    notificationCount,
  };
}

export function checkNewNotificationsErrorAction(error) {
  return {
    type: CHECK_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function getNewNotificationsAction() {
  return {
    type: GET_NEW_NOTIFICATIONS,
  };
}

export function getNewNotificationsSuccessAction(notifications) {
  return {
    type: GET_NEW_NOTIFICATIONS_SUCCESS,
    notifications,
  };
}

export function getNewNotificationsErrorAction(error) {
  return {
    type: GET_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function unsetNewNotificationsAction() {
  return {
    type: UNSET_NEW_NOTIFICATIONS,
  };
}

export function unsetNewNotificationsSuccessAction() {
  return {
    type: UNSET_NEW_NOTIFICATIONS_SUCCESS,
  };
}

export function unsetNewNotificationsErrorAction(error) {
  return {
    type: UNSET_NEW_NOTIFICATIONS_ERROR,
    error,
  };
}

export function isLoggedAction() {
  return {
    type: IS_LOGGED,
  };
}

export function loggedInAction(userId, token) {
  return {
    type: LOGGED_IN,
    userId,
    token,
  };
}

export function logoutAction() {
  return {
    type: LOGOUT,
  };
}

export function logoutSuccessAction() {
  return {
    type: LOGOUT_SUCCESS,
  };
}

export function toggleNavigationDesktopAction() {
  return {
    type: TOGGLE_NAVIGATION_DESKTOP,
  };
}

export function toggleNavigationMobileAction() {
  return {
    type: TOGGLE_NAVIGATION_MOBILE,
  };
}

export function toggleMessagesAction() {
  return {
    type: TOGGLE_MESSAGES,
  };
}

export function toggleNotificationsAction() {
  return {
    type: TOGGLE_NOTIFICATIONS,
  };
}

export function logoutErrorAction(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}

export function enqueueSnackbarAction(snackbar) {
  const key = snackbar.options && snackbar.options.key;

  return {
    type: ENQUEUE_SNACKBAR,
    snackbar: {
      ...snackbar,
      key: key || new Date().getTime() + Math.random(),
    },
  };
}

export function closeSnackbarAction(key) {
  return {
    type: CLOSE_SNACKBAR,
    dismissAll: !key,
    key,
  };
}

export function removeSnackbarAction(key) {
  return {
    type: REMOVE_SNACKBAR,
    key,
  };
}
