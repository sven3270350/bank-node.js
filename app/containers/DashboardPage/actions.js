/*
 *
 * DashboardPage actions
 *
 */

import {
  GET_NAME,
  GET_NAME_SUCCESS,
  GET_NAME_ERROR,
  GET_SURNAME,
  GET_SURNAME_SUCCESS,
  GET_SURNAME_ERROR,
  GET_EMAIL,
  GET_EMAIL_SUCCESS,
  GET_EMAIL_ERROR,
  GET_LAST_PRESENT_LOGGED,
  GET_LAST_PRESENT_LOGGED_SUCCESS,
  GET_LAST_PRESENT_LOGGED_ERROR,
  GET_LAST_SUCCESSFUL_LOGGED,
  GET_LAST_SUCCESSFUL_LOGGED_SUCCESS,
  GET_LAST_SUCCESSFUL_LOGGED_ERROR,
  GET_LAST_FAILED_LOGGED,
  GET_LAST_FAILED_LOGGED_SUCCESS,
  GET_LAST_FAILED_LOGGED_ERROR,
  GET_AVAILABLE_FUNDS,
  GET_AVAILABLE_FUNDS_SUCCESS,
  GET_AVAILABLE_FUNDS_ERROR,
  GET_ACCOUNT_BILLS,
  GET_ACCOUNT_BILLS_SUCCESS,
  GET_ACCOUNT_BILLS_ERROR,
  GET_ACCOUNT_BALANCE_HISTORY,
  GET_ACCOUNT_BALANCE_HISTORY_SUCCESS,
  GET_ACCOUNT_BALANCE_HISTORY_ERROR,
  GET_SAVINGS,
  GET_SAVINGS_SUCCESS,
  GET_SAVINGS_ERROR,
  GET_RECENT_TRANSACTIONS_SENDER,
  GET_RECENT_TRANSACTIONS_SENDER_SUCCESS,
  GET_RECENT_TRANSACTIONS_SENDER_ERROR,
  GET_RECENT_TRANSACTIONS_RECIPIENT,
  GET_RECENT_TRANSACTIONS_RECIPIENT_SUCCESS,
  GET_RECENT_TRANSACTIONS_RECIPIENT_ERROR,
  GET_OUTGOING_TRANSFERS_SUM,
  GET_OUTGOING_TRANSFERS_SUM_SUCCESS,
  GET_OUTGOING_TRANSFERS_SUM_ERROR,
  GET_INCOMING_TRANSFERS_SUM,
  GET_INCOMING_TRANSFERS_SUM_SUCCESS,
  GET_INCOMING_TRANSFERS_SUM_ERROR,
  GET_CURRENCY,
  GET_CURRENCY_SUCCESS,
  GET_CURRENCY_ERROR,
  GET_CURRENCY_ID,
  GET_CURRENCY_ID_SUCCESS,
  GET_CURRENCY_ID_ERROR,
  GET_RECHARTS_DATA,
  GET_RECHARTS_DATA_SUCCESS,
  GET_RECHARTS_DATA_ERROR,
  GET_RECHARTS_COLORS,
  GET_RECHARTS_COLORS_SUCCESS,
  GET_RECHARTS_COLORS_ERROR,
} from './constants';

export function getNameAction() {
  return {
    type: GET_NAME,
  };
}

export function getNameSuccessAction(name) {
  return {
    type: GET_NAME_SUCCESS,
    name,
  };
}

export function getNameErrorAction(error) {
  return {
    type: GET_NAME_ERROR,
    error,
  };
}

export function getSurnameAction() {
  return {
    type: GET_SURNAME,
  };
}

export function getSurnameSuccessAction(surname) {
  return {
    type: GET_SURNAME_SUCCESS,
    surname,
  };
}

export function getSurnameErrorAction(error) {
  return {
    type: GET_SURNAME_ERROR,
    error,
  };
}

export function getEmailAction() {
  return {
    type: GET_EMAIL,
  };
}

export function getEmailSuccessAction(email) {
  return {
    type: GET_EMAIL_SUCCESS,
    email,
  };
}

export function getEmailErrorAction(error) {
  return {
    type: GET_EMAIL_ERROR,
    error,
  };
}

export function getCurrencyAction() {
  return {
    type: GET_CURRENCY,
  };
}

export function getCurrencySuccessAction(currency) {
  return {
    type: GET_CURRENCY_SUCCESS,
    currency,
  };
}

export function getCurrencyErrorAction(error) {
  return {
    type: GET_CURRENCY_ERROR,
    error,
  };
}

export function getCurrencyIdAction() {
  return {
    type: GET_CURRENCY_ID,
  };
}

export function getCurrencyIdSuccessAction(currencyId) {
  return {
    type: GET_CURRENCY_ID_SUCCESS,
    currencyId,
  };
}

export function getCurrencyIdErrorAction(error) {
  return {
    type: GET_CURRENCY_ID_ERROR,
    error,
  };
}

export function getLastPresentLoggedAction() {
  return {
    type: GET_LAST_PRESENT_LOGGED,
  };
}

export function getLastPresentLoggedSuccessAction(lastPresentLogged) {
  return {
    type: GET_LAST_PRESENT_LOGGED_SUCCESS,
    lastPresentLogged,
  };
}

export function getLastPresentLoggedErrorAction(error) {
  return {
    type: GET_LAST_PRESENT_LOGGED_ERROR,
    error,
  };
}

export function getLastSuccessfulLoggedAction() {
  return {
    type: GET_LAST_SUCCESSFUL_LOGGED,
  };
}

export function getLastSuccessfulLoggedSuccessAction(lastSuccessfulLogged) {
  return {
    type: GET_LAST_SUCCESSFUL_LOGGED_SUCCESS,
    lastSuccessfulLogged,
  };
}

export function getLastSuccessfulLoggedErrorAction(error) {
  return {
    type: GET_LAST_SUCCESSFUL_LOGGED_ERROR,
    error,
  };
}

export function getLastFailedLoggedAction() {
  return {
    type: GET_LAST_FAILED_LOGGED,
  };
}

export function getLastFailedLoggedSuccessAction(lastFailedLogged) {
  return {
    type: GET_LAST_FAILED_LOGGED_SUCCESS,
    lastFailedLogged,
  };
}

export function getLastFailedLoggedErrorAction(error) {
  return {
    type: GET_LAST_FAILED_LOGGED_ERROR,
    error,
  };
}

export function getAvailableFundsAction() {
  return {
    type: GET_AVAILABLE_FUNDS,
  };
}

export function getAvailableFundsSuccessAction(availableFunds) {
  return {
    type: GET_AVAILABLE_FUNDS_SUCCESS,
    availableFunds,
  };
}

export function getAvailableFundsErrorAction(error) {
  return {
    type: GET_AVAILABLE_FUNDS_ERROR,
    error,
  };
}

export function getAccountBillsAction() {
  return {
    type: GET_ACCOUNT_BILLS,
  };
}

export function getAccountBillsSuccessAction(accountBills) {
  return {
    type: GET_ACCOUNT_BILLS_SUCCESS,
    accountBills,
  };
}

export function getAccountBillsErrorAction(error) {
  return {
    type: GET_ACCOUNT_BILLS_ERROR,
    error,
  };
}

export function getSavingsAction() {
  return {
    type: GET_SAVINGS,
  };
}

export function getSavingsSuccessAction(savings) {
  return {
    type: GET_SAVINGS_SUCCESS,
    savings,
  };
}

export function getSavingsErrorAction(error) {
  return {
    type: GET_SAVINGS_ERROR,
    error,
  };
}

export function getAccountBalanceHistoryAction() {
  return {
    type: GET_ACCOUNT_BALANCE_HISTORY,
  };
}

export function getAccountBalanceHistorySuccessAction(accountBalanceHistory) {
  return {
    type: GET_ACCOUNT_BALANCE_HISTORY_SUCCESS,
    accountBalanceHistory,
  };
}

export function getAccountBalanceHistoryErrorAction(error) {
  return {
    type: GET_ACCOUNT_BALANCE_HISTORY_ERROR,
    error,
  };
}

export function getRecentTransactionsSenderAction() {
  return {
    type: GET_RECENT_TRANSACTIONS_SENDER,
  };
}

export function getRecentTransactionsSenderSuccessAction(
  recentTransactionsSender,
) {
  return {
    type: GET_RECENT_TRANSACTIONS_SENDER_SUCCESS,
    recentTransactionsSender,
  };
}

export function getRecentTransactionsSenderErrorAction(error) {
  return {
    type: GET_RECENT_TRANSACTIONS_SENDER_ERROR,
    error,
  };
}

export function getRecentTransactionsRecipientAction() {
  return {
    type: GET_RECENT_TRANSACTIONS_RECIPIENT,
  };
}

export function getRecentTransactionsRecipientSuccessAction(
  recentTransactionsRecipient,
) {
  return {
    type: GET_RECENT_TRANSACTIONS_RECIPIENT_SUCCESS,
    recentTransactionsRecipient,
  };
}

export function getRecentTransactionsRecipientErrorAction(error) {
  return {
    type: GET_RECENT_TRANSACTIONS_RECIPIENT_ERROR,
    error,
  };
}

export function getIncomingTransfersSumAction() {
  return {
    type: GET_INCOMING_TRANSFERS_SUM,
  };
}

export function getIncomingTransfersSumSuccessAction(incomingTransfersSum) {
  return {
    type: GET_INCOMING_TRANSFERS_SUM_SUCCESS,
    incomingTransfersSum,
  };
}

export function getIncomingTransfersSumErrorAction(error) {
  return {
    type: GET_INCOMING_TRANSFERS_SUM_ERROR,
    error,
  };
}

export function getOutgoingTransfersSumAction() {
  return {
    type: GET_OUTGOING_TRANSFERS_SUM,
  };
}

export function getOutgoingTransfersSumSuccessAction(outgoingTransfersSum) {
  return {
    type: GET_OUTGOING_TRANSFERS_SUM_SUCCESS,
    outgoingTransfersSum,
  };
}

export function getOutgoingTransfersSumErrorAction(error) {
  return {
    type: GET_OUTGOING_TRANSFERS_SUM_ERROR,
    error,
  };
}

export function getRechartsDataAction() {
  return {
    type: GET_RECHARTS_DATA,
  };
}

export function getRechartsDataSuccessAction(rechartsData) {
  return {
    type: GET_RECHARTS_DATA_SUCCESS,
    rechartsData,
  };
}

export function getRechartsDataErrorAction(error) {
  return {
    type: GET_RECHARTS_DATA_ERROR,
    error,
  };
}

export function getRechartsColorsAction() {
  return {
    type: GET_RECHARTS_COLORS,
  };
}

export function getRechartsColorsSuccessAction(rechartsColors) {
  return {
    type: GET_RECHARTS_COLORS_SUCCESS,
    rechartsColors,
  };
}

export function getRechartsColorsErrorAction(error) {
  return {
    type: GET_RECHARTS_COLORS_ERROR,
    error,
  };
}
