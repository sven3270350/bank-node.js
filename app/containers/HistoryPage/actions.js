/*
 *
 * HistoryPage actions
 *
 */

import {
  GRID_STATE_CHANGE,
  GET_GRID_DATA,
  GET_GRID_DATA_SUCCESS,
  GET_GRID_DATA_TRANSFORM_SUCCESS,
  GET_GRID_DATA_ERROR,
} from './constants';

export function createGridAction(partialStateName, partialStateValue) {
  return {
    type: GRID_STATE_CHANGE,
    partialStateName,
    partialStateValue,
  };
}

export function getGridDataAction() {
  return {
    type: GET_GRID_DATA,
  };
}

export function getGridDataSuccessAction() {
  return {
    type: GET_GRID_DATA_SUCCESS,
  };
}

export function getGridDataErrorAction() {
  return {
    type: GET_GRID_DATA_ERROR,
  };
}

export function getGridDataTransformSuccessAction(rowsTransform) {
  return {
    type: GET_GRID_DATA_TRANSFORM_SUCCESS,
    rowsTransform,
  };
}
