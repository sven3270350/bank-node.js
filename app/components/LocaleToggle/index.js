/*
 *
 * LanguageToggle
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import { changeLocale } from 'containers/LanguageProvider/actions';
import { makeSelectLocale } from 'containers/LanguageProvider/selectors';
import Toggle from './Toggle';
import messages from './messages';
import { appLocales } from '../../i18n';

export function LocaleToggle(props) {
  return (
    <div>
      <Toggle
        value={props.locale}
        values={appLocales}
        messages={messages}
        onToggle={props.onLocaleToggle}
      />
    </div>
  );
}

LocaleToggle.propTypes = {
  onLocaleToggle: PropTypes.func,
  locale: PropTypes.string,
};

const mapStateToProps = createSelector(
  makeSelectLocale(),
  locale => ({
    locale,
  }),
);

export function mapDispatchToProps(dispatch) {
  return {
    onLocaleToggle: evt => dispatch(changeLocale(evt.target.value)),
    dispatch,
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LocaleToggle);
