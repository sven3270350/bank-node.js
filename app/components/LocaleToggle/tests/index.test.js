import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';

import { changeLocale } from 'containers/LanguageProvider/actions';
import LanguageProvider from 'containers/LanguageProvider';
import LocaleToggle, { mapDispatchToProps } from '../index';

import configureStore from '../../../configureStore';
import { translationMessages } from '../../../i18n';

describe('<LocaleToggle />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should present the default `en` english language option', () => {
    const { container } = render(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <LocaleToggle />
        </LanguageProvider>
      </Provider>,
    );
    expect(container.querySelector('option[value="en"]')).not.toBeNull();
  });

  describe('mapDispatchToProps', () => {
    describe('onLocaleToggle', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLocaleToggle).toBeDefined();
      });

      it('should dispatch changeLocale when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const locale = 'de';
        const evt = { target: { value: locale } };
        result.onLocaleToggle(evt);
        expect(dispatch).toHaveBeenCalledWith(changeLocale(locale));
      });
    });
  });
});
