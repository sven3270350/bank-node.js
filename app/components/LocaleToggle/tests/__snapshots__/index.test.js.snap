// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocaleToggle /> should match the snapshot 1`] = `
<div>
  <select
    class="Select-sc-15n6sxn-0 btfomW"
  >
    <option
      value="en"
    >
      EN
    </option>
  </select>
</div>
`;
